### PR TITLE
chore(js): add expo example app and move to pnpm

### PR DESCRIPTION
**Pull request by berendsliedrecht**
- supersedes #246
- builds on top of #245
- Moves to pnpm from yarn
- Update lerna from 7 to 8 to fix issue with building
- Added react native (expo) example app

## Rust side changes
- Updated the secure-env library as there were some issues with Android in a non `cargo-ndk` managed app.
- Added a `HasKeyBackend` trait which defaults to `KeyBackend::Software`. This is done so we can correctly create P256 Software vs Hardware keys.

### CODE REVIEW

**Comment by berendsliedrecht** [comment]
@swcurran snyk seems to fail, but I cannot access the project page on their website. Could you help with this? I just need to know which dependencies are the issue.

**Comment by swcurran** [comment]
@berendsliedrecht — is this helpfuil?  Not sure it is what you want:

image

**Comment by berendsliedrecht** [comment]
> @berendsliedrecht — is this helpfuil?  Not sure it is what you want. It what I can see, and the only related link is the settings gear, but that doesn’t seem to be helpful.
> 
> 
> 
> image
> 
> 

Hmm it wants it to be built with yarn. Do we have some setting to set the package manager? Pnpm-lock.yaml is equal to yarn.lock

**Comment by swcurran** [comment]
I don’t know about any of this stuff, so stumbling around.  When I hit the “gear” from the image above, and then in the top right “Open in GitHub” — it takes me to this page: https://github.com/hyperledger/aries-askar/blob/main/wrappers/javascript/package.json where there are a number of references to yarn. Is that file correct?

**Comment by TimoGlastra** [comment]
Requested us to be added @berendsliedrecht: https://github.com/hyperledger/governance/issues/259

**Review by TimoGlastra — DISMISSED** [review]
LGTM, nice 👍

File: wrappers/javascript/.eslintrc.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+root: true
```
[the remark this answers is not in the document]
Comment (berendsliedrecht) [reply wrappers/javascript/.eslintrc.yaml:1]: Yes, wanted to do that at a later stage.

**Comment by berendsliedrecht** [comment]
Going a bit through the Snyk issue again, it seems that they do not have support for PNPM (https://github.com/snyk/nodejs-lockfile-parser/issues/111). Can we disable it for the JS projects and just have dependabot deal with vunelerabilities?

**Comment by swcurran** [comment]
@WadeBarnes — can you help here?  You have a way better handle on this than I.  Thanks!

**Comment by swcurran** [comment]
@WadeBarnes — I mean about the Snyk comment — https://github.com/hyperledger/aries-askar/pull/247#issuecomment-2112301048

**Comment by WadeBarnes** [comment]
It seems Snyk is adding support for pnpm; https://github.com/snyk/nodejs-lockfile-parser/issues/111#issuecomment-2060852547

Some additional questions to establish some background (mostly for me since the setup is not obvious):
- How does Snyk come into play with what's being discussed here?
- How is Snyk integrated with the project?
  - At what level?
  - Directly or indirectly?

**Review by TimoGlastra — APPROVED** [review]
LGTM except for pushing android/ios folders for example app

**Review by TimoGlastra — APPROVED** [review]

**Comment by TimoGlastra** [comment]
DCO seems to be messed up. Maybe you need to squash and force push?

**Comment by berendsliedrecht** [comment]
@WadeBarnes 

> Some additional questions to establish some background (mostly for me since the setup is not obvious):

> How does Snyk come into play with what's being discussed here?
> How is Snyk integrated with the project?
> At what level?
> Directly or indirectly?

Sadly I am not a Snyk expert so I cannot answer these questions. I think it is integrated indirectly just by being under the Hyperledger organization (not a 100% sure, but I cannot seem to find any Snyk setup in this repo). IMO we can just disable and let dependabot deal with the specific issues for JS. This is blocking quite some work now, also for Credo.

**Comment by WadeBarnes** [comment]
@ryjones, Do you have any thoughts, comments, or recommendations on the Snyk topic?
- https://github.com/hyperledger/aries-askar/pull/247#issuecomment-2112301048
- https://github.com/hyperledger/aries-askar/pull/247#issuecomment-2112787790
- https://github.com/hyperledger/aries-askar/pull/247#issuecomment-2154335671

**Comment by ryjones** [comment]
what is weird is I only see one repo on the allow list

**Comment by TimoGlastra** [comment]
@berendsliedrecht Made a similar PR in indy vdr repo, but also updated to Biome and vitest there (lest config/setup). Maybe we can do the same here as well.

**Pull request merged**